From 34585fd2a075ca52cf1f29ab72324354dbd72889 Mon Sep 17 00:00:00 2001
From: Kaifeng Zhu <kaifeng.zhu@citrix.com>
Date: Fri, 7 Mar 2014 15:27:36 +0000
Subject: [PATCH 05/10] qemu-char: fix memory leak in qemu_char_open_pty()
To: andrew.cooper3@citrix.com

The momery pointed by s and chr could be leaked if openpty return a value
less then 0.
Coverity issue id: 22453.

Signed-off-by: Kaifeng Zhu <kaifeng.zhu@citrix.com>
diff --git a/qemu-char.c b/qemu-char.c
index 6acaa8c..b541dd3 100644
--- a/qemu-char.c
+++ b/qemu-char.c
@@ -932,6 +932,8 @@ static CharDriverState *qemu_chr_open_pty(void)
     s = qemu_mallocz(sizeof(PtyCharDriver));
 
     if (openpty(&s->fd, &slave_fd, pty_name, NULL, NULL) < 0) {
+        qemu_free(s);
+        qemu_free(chr);
         return NULL;
     }
 
